contributing.md

```diff
@@ -187,6 +191,107 @@ conda create \
         legate-boost
 ```
 
+## Releasing
+
+NOTE: some steps in this section require direct write access to the repo (including its `main` branch).
+
+### Create a stable release
+
+1. Create a pull request updating the `VERSION` file on the `main` branch to the desired version, with no leading `v`
+
+```shell
+echo "24.09.00" > ./VERSION
```

Comment:
Right, it's expected to be found via the tag.

This is what RAPIDS does, see https://github.com/rapidsai/cudf/tags

<img width="821" alt="Screenshot 2024-10-14 at 10 28 19 AM" src="https://github.com/user-attachments/assets/491895a2-4976-4f5a-8b88-3dbfb88dbe05">

And it's how `legate` is doing it too: https://github.com/nv-legate/legate.core.internal/tags

<img width="859" alt="Screenshot 2024-10-14 at 10 29 36 AM" src="https://github.com/user-attachments/assets/821eae42-13d6-4a0d-97ee-14ff93d914bc">

Sorry, I'm struggling today to remember exactly why that's preferable to including the `dev` part in the file or in the code that calculates "number of commits since the last tag" to generate a version like `24.12.00dev5`.